chat-plugins/roomsettings.js

```diff
@@ -0,0 +1,131 @@
+			} else if (RANKS[i]) {
+				// Personal rooms modjoin check
+				if (this.room.isPersonal && !this.user.can('makeroom') && RANKS[i] !== '+') continue;
+
+				modjoinOutput.push('<button name="send" value="/roomsetting modjoin ' + RANKS[i] + '">' + RANKS[i] + '</button>');
+			}
+		}
+		return modjoinOutput.join(" | ");
+	}
+	stretching() {
+		if (!this.user.can('editroom', null, this.room)) return "<button " + DISABLED + ">" + (this.room.stretching ? this.room.stretching : false) + "</button>";
+		if (this.room.filterStretching) {
+			return '<button name="send" value="/roomsetting stretching disable">false</button> | <button ' + DISABLED + '>true</button>';
+		} else {
+			return '<button ' + DISABLED + '>false</button> | <button name="send" value="/roomsetting stretching enable">true</button>';
+		}
+	}
+	capitals() {
+		if (!this.user.can('editroom', null, this.room)) return "<button " + DISABLED + ">" + (this.room.capitals ? this.room.capitals : false) + "</button>";
+		if (this.room.filterCaps) {
+			return '<button name="send" value="/roomsetting capitals disable">false</button> | <button ' + DISABLED + '>true</button>';
+		} else {
+			return '<button ' + DISABLED + '>false</button> | <button name="send" value="/roomsetting capitals enable">true</button>';
+		}
+	}
+	slowchat() {
+		if (!this.user.can('editroom', null, this.room) || this.room.userCount < SLOWCHAT_USER_REQUIREMENT) return "<button " + DISABLED + ">" + (this.room.slowchat ? this.room.slowchat : false) + "</button>";
+
+		let slowchatOutput = [];
+		for (let i = 10; i <= 60; i += 10) {
+			if (this.room.slowchat === i) {
+				slowchatOutput.push('<button ' + DISABLED + '>' + i + '</button>');
+			} else {
+				slowchatOutput.push('<button name="send" value="/roomsetting slowchat ' + i + '">' + i + '</button>');
+			}
+		}
+		if (!this.room.slowchat) {
+			slowchatOutput.unshift('<button ' + DISABLED + '>false</button>');
+		} else {
+			slowchatOutput.unshift('<button name="send" value="/roomsettings slowchat false">false</button>');
+		}
+		return slowchatOutput.join(" | ");
+	}
+	generateDisplay(user, room, connection) {
+		let output = '<div class="infobox">Room Settings for ' + Tools.escapeHTML(this.room.title) + '<br />';
+		output += "<u>Modchat:</u> " + this.modchat() + "<br />";
+		output += "<u>Modjoin:</u> " + this.modjoin() + "<br />";
+		output += "<u>Stretching:</u> " + this.stretching() + "<br />";
+		output += "<u>Capitals:</u> " + this.capitals() + "<br />";
+		output += "<u>Slowchat:</u> " + this.slowchat() + "<br />";
+		output += "</div>";
+
+		this.user.sendTo(this.room, '|uhtml' + (this.sameCommand ? '' : 'change') + '|roomsettings|' + output);
+	}
+}
+
+exports.commands = {
+	roomsetting: 'roomsettings',
+	roomsettings: function (target, room, user, connection) {
+		if (room.battle) return this.errorReply("This command cannot be used in battle rooms.");
+		room.settings = new RoomSettings(user, room, connection);
```

Comment:
I did this so that if they just used the command, it will know when to update.  (see ``this.sameCommand``)